src/coreclr/tools/Common/Compiler/ObjectWriter/WasmObjectWriter.cs

```diff
@@ -0,0 +1,233 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using ILCompiler.DependencyAnalysis;
+using ILCompiler.DependencyAnalysisFramework;
+using Internal.TypeSystem;
+
+namespace ILCompiler.ObjectWriter
+{
+    // This is a placeholder for now. It will need to be filled in with ABI-specific signatures
+    // for any method we want to emit into the Wasm module.
+    public class WasmAbiContext
```

Comment:
What is the information that this is going to hold? `MethodDesc -> WasmFuncType` is a pure function, it will have access to all the ABI parameters (if we decide to introduce any) from the type system context.

Reply:
Ah, that is good to know that `MethodDesc -> WasmFuncType` is a pure operation. I wasn't sure. I was thinking that if crossgen needs to compute that mapping, the results could be cached in a dictionary or something like that here. I'll change this to a static class for now though and remove it as an instance field for the object writer.